.vib/rabbitmq/cypress/cypress/integration/rabbitmq_spec.js

```diff
@@ -0,0 +1,102 @@
+});
+
+it('allows publishing a message to a created exchange', () => {
+  cy.login();
+  cy.visit('#/exchanges');
+  cy.contains('Add a new exchange').click();
+  cy.fixture('exchanges').then((exchange) => {
+    cy.get('[name="name"]')
+      .scrollIntoView()
+      .type(`${exchange.newExchange.name}${random}`);
+    cy.get('#arguments_1_mfkey').type(exchange.newExchange.argument1);
+    cy.get('#arguments_1_mfvalue').type(exchange.newExchange.argument2);
+    cy.contains('Add exchange').click({ force: true });
+    cy.get('table[class="list"]').should(
+      'contain',
+      `${exchange.newExchange.name}${random}`
+    );
+    cy.contains('a', `${exchange.newExchange.name}${random}`).click();
+  });
+  cy.contains('Publish message').click();
+  cy.fixture('messages').then((message) => {
+    cy.get('textarea').type(message.newMessage.payload);
+  });
+  cy.contains('input', 'Publish message').click();
+  cy.contains('Message published, but not routed');
+});
+
+it('allows adding a new queue and binding/unbinding it to the exchange', () => {
+  cy.login();
+  cy.visit('#/queues');
+  cy.contains('Add a new queue').click();
+  cy.fixture('queues').then((queue) => {
+    cy.get('[name="name"]').type(`${queue.newQueue.name}${random}`);
+    cy.get('#arguments_1_mfkey').type(queue.newQueue.argument1);
+    cy.get('#arguments_1_mfvalue').type(queue.newQueue.argument2);
+    cy.contains('Add queue').click();
+    cy.get('table[class="list"]').should(
+      'contain',
+      `${queue.newQueue.name}${random}`
+    );
+  });
+  cy.visit('#/exchanges');
+  cy.contains('amq.direct').click();
+  cy.contains('Bindings').click();
+  cy.fixture('queues').then((queue) => {
+    cy.get('[name="destination"]').type(`${queue.newQueue.name}${random}`);
+    cy.contains('[type="submit"]', 'Bind').click();
+    cy.contains('.bindings-wrapper', `${queue.newQueue.name}${random}`);
+  });
+  cy.contains('Unbind').click();
+  cy.contains('no bindings');
+});
+
+it('allows adding a new admin and logging in as such', () => {
+  const APP_NAME = 'RabbitMQ';
+
+  cy.login();
+  cy.visit('/#/users');
+  cy.contains('Add a user').click();
+  cy.fixture('admins').then((admin) => {
+    cy.get('[name="username"').type(`${admin.newAdmin.userName}${random}`);
+    cy.get('[name="password"]').type(`${admin.newAdmin.password}${random}`);
+    cy.get('[name="password_confirm"]').type(
+      `${admin.newAdmin.password}${random}`
+    );
+    cy.get('#tags').type(admin.newAdmin.tags);
+    cy.contains('input[type="submit"]', 'Add user').click();
+    cy.get('[class="updatable"]').should(
+      'contain',
+      `${admin.newAdmin.userName}${random}`
+    );
+    cy.contains('Log out').click();
+    cy.get('[alt="RabbitMQ logo"]');
+    cy.get('[name="username"]').type(`${admin.newAdmin.userName}${random}`);
+    cy.get('[name="password"]').type(`${admin.newAdmin.password}${random}`);
+    cy.contains('Login').click();
+    cy.get('#versions').should('contain', APP_NAME);
```

Comment:
I'm logging in with **different** credentials this time (the newly created admin).
Since the Cypress custom login method doesn't accept arguments as far as I can see, I can't use it. 
Do you think I should modify the login method to accept arguments and then pass admin credentials?

Reply:
The custom method does accept arguments, but defaults to the `cypress.json` defined username/password:

```javascript
Cypress.Commands.add(
  'login',
  (username = Cypress.env('username'), password = Cypress.env('password')) => {
```

You can provide your own values, though 😁 

```javascript
cy.login('myuser', 'mypass');
```